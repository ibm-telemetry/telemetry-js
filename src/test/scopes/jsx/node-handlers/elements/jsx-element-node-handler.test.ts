/*
 * Copyright IBM Corp. 2023, 2023
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { describe, it } from 'vitest'

describe('class: JsxElementNodeHandler', () => {
  it.todo('correctly returns JSXElement when prefix is present')
  it.todo('correctly returns JSXElement when prefix is absent')
  it.todo('correctly returns JSXElement with deeply referenced element (a.b.c.d)')
  it.todo('correctly returns attributes on the JSXElement')
})
