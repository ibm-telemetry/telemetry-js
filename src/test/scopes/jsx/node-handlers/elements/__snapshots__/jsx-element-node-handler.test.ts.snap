// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`class: JsxElementNodeHandler > correctly returns the JsxElements for a complex fixture 1`] = `
[
  {
    "attributes": [],
    "name": "Element1",
    "prefix": "Prefix",
    "raw": "<Prefix.Element1>SomethingHere</Prefix.Element1>",
  },
  {
    "attributes": [],
    "name": "Element1",
    "prefix": undefined,
    "raw": "<Element1>SomethingHere</Element1>",
  },
  {
    "attributes": [],
    "name": "Property.Nested.Element",
    "prefix": "Prefix",
    "raw": "<Prefix.Property.Nested.Element>SomethingHere</Prefix.Property.Nested.Element>",
  },
  {
    "attributes": [
      {
        "name": "attr1",
        "value": "\\"string\\"",
      },
      {
        "name": "attr2",
        "value": "4",
      },
      {
        "name": "attr5",
        "value": "undefined",
      },
      {
        "name": "attr6",
        "value": "null",
      },
      {
        "name": "attr9",
        "value": "true",
      },
      {
        "name": "attr10",
        "value": "false",
      },
      {
        "name": "attr11",
        "value": "variable",
      },
      {
        "name": "attr12",
        "value": "element[access]",
      },
      {
        "name": "attr13",
        "value": "{key: 'value', key2: 10}",
      },
    ],
    "name": "Element1",
    "prefix": undefined,
    "raw": "<Element1
        attr1=\\"string\\"
        attr2={4}
        attr5={undefined}
        attr6={null}
        attr8
        attr9={true}
        attr10={false}
        attr11={variable}
        attr12={element[access]}
        attr13={{key: 'value', key2: 10}}
        {...spreadObj}
        exp={}>
        SomethingHere
    </Element1>",
  },
]
`;
